### DISCUSSION TITLE
Local VCS file deleted, how to autocreate missing files?

### DISCUSSION THREAD

**Discussion by goranZivkovicAbel** [WeblateOrg/weblate#9632] [Q&A] [answered]
Hi all,

Weblate docker version: 4.18.2.1

I have local VCS setup in i18n json format. All translates are stored in {lang}.json format and later pulled by third party service. Maybe not best solution but it works.
I tested disaster recovery scenario where I deleted docker volume, result are numerous exceptions from scheduled tasks that say File not found. Also when try to Download translation file using REST API it returns File not found and by source code this is expected to happen.

Is there way to prevent this and allow weblate to create file if missing?

Bonus question is do we have option to use redis instead disk reading, or custom implementation is needed?

Thanks in advance and sorry if topic is already covered!

![image](https://github.com/WeblateOrg/weblate/assets/97457572/17e2f276-f737-4061-b6e6-d161fa11dd57)

**Comment by nijel** [comment]
No, the file system is needed.

You should have backups for disaster recovery, see https://docs.weblate.org/en/latest/admin/backup.html

**Reply by goranZivkovicAbel** [reply]
What if .json file is created and we run commit again. Will that populate file with data again?

**Reply by nijel** [reply, marked as answer]
You can force sync in the repository maintenance.

**Reply by goranZivkovicAbel** [reply]
> You can force sync in the repository maintenance.

Thanks a lot for help
